components/RecentlyPlayed.tsx

```diff
@@ -0,0 +1,846 @@
+    Appearance.setColorScheme('dark');
+
+    return () => {
+      // Stop any playing preview when component unmounts
+      audioPreviewService.stop();
+      // Reset appearance
+      Appearance.setColorScheme(null);
+    };
+  }, [audioPreviewService]);  
+
+  useEffect(() => {
+    // Subscribe to preview state changes
+    const unsubscribe = audioPreviewService.subscribe(setPreviewState);
+    return unsubscribe;
+  }, [audioPreviewService]);
+
+  // subscribes to RecentlyPlayedService subscriber
+  useEffect(() => {
+  const unsubscribe = recentlyPlayedService.subscribeToCurrentShow((show) => {
+    setCurrentShow(show ?? undefined);
+  });
+  return unsubscribe;
+}, [recentlyPlayedService]);
+
+  const parsePlaylistTimestamp = (timeStr: string): Date => {
+    try {
+      // Format: YYYY/MM/DD HH:MM:SS
+      const [datePart, timePart] = timeStr.split(' ');
+      
+      if (!datePart || !timePart) {
+        return new Date();
+      }
+      
+      const [year, month, day] = datePart.split('/').map(Number);
+      const [hour, minute, second] = timePart.split(':').map(Number);
+      
+      return new Date(year, month - 1, day, hour, minute, second);
+    } catch (parseError) {
+      return new Date();
+    }
+  };
+
+  const fetchShowPlaylist = useCallback(async (showName: string, date: string): Promise<ProcessedSong[]> => {
+    const encodedShowName = encodeURIComponent(showName);
+    const url = `https://wmbr.alexandersimoes.com/get_playlist?show_name=${encodedShowName}&date=${date}`;
+    
+    const response = await fetch(url, {
+      headers: { 'Cache-Control': 'no-cache' }
+    });
+    
+    if (!response.ok) {
+      // If it's a 404, return empty list instead of throwing error
+      if (response.status === 404) {
+        return [];
+      }
+      
+      throw new Error(`Failed to fetch playlist: ${response.status}`);
+    }
+    
+    const playlistData = await response.json();
+    
+    // If the response has an "error" key, return empty list
+    if (playlistData.error) {
+      return [];
+    }
+    
+    if (playlistData.songs && playlistData.songs.length > 0) {
+      // Convert playlist songs to ProcessedSong format
+      const processedSongs: ProcessedSong[] = playlistData.songs.map((song: any) => ({
+        title: song.song.trim(),
+        artist: song.artist.trim(),
+        album: song.album?.trim() || undefined,
+        released: undefined,
+        appleStreamLink: '', // Not provided in new API
+        playedAt: parsePlaylistTimestamp(song.time),
+        showName: showName,
+        showId: `${showName}-${date}`
+      }));
+      
+      // Sort by most recent first
+      processedSongs.sort((a, b) => b.playedAt.getTime() - a.playedAt.getTime());
+      return processedSongs;
+    }
+    
+    return [];
+  }, []);
+
+  const fetchCurrentShowPlaylist = useCallback(async (isRefresh = false) => {
+    if (!currentShow || currentShow === 'WMBR 88.1 FM') {
```

Comment:
AFAICT this isn't debouncing fetches/returning early if loading or refreshing - we should do so since otherwise it'll be triggering extra requests that can't succeed.

plus in local dev - react runs all hooks twice, so this may be causing some local dev weirdness